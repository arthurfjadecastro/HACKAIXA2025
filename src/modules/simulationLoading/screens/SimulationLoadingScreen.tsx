import React, { useState, useEffect } from 'react';
import { View, SafeAreaView } from 'react-native';
import { useRoute, RouteProp, useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { AppStackParamList } from '@/navigation/AppStack';
import { useFinancialCalculation } from '@/modules/simulations/hooks/useFinancialCalculation';

import SimulationLottieAnimation from '../components/LottieAnimation';
import { styles } from './SimulationLoadingScreen.styles';

type RouteProps = RouteProp<AppStackParamList, 'SimulationLoading'>;
type NavigationProps = NativeStackNavigationProp<AppStackParamList>;

type LoadingStatus = 'loading' | 'success';

const SimulationLoadingScreen: React.FC = () => {
  const route = useRoute<RouteProps>();
  const navigation = useNavigation<NavigationProps>();
  const [status, setStatus] = useState<LoadingStatus>('loading');

  // Usar o hook de c√°lculo financeiro
  const { calculation } = useFinancialCalculation({
    productId: route.params?.productId || '',
    amount: route.params?.amount || '0',
    months: route.params?.months || 96
  });

  // Simula o processamento da simula√ß√£o
  useEffect(() => {
    const simulateCalculation = async () => {
      console.log('üßÆ Iniciando c√°lculo da simula√ß√£o...');
      
      const calculationTime = 4000; // 4 segundos
      
      setTimeout(() => {
        console.log('‚úÖ C√°lculo da simula√ß√£o conclu√≠do');
        setStatus('success');
      }, calculationTime);
    };

    simulateCalculation();
  }, []);

  const handleAnimationFinish = async () => {
    console.log('üé¨ Anima√ß√£o finalizada - Status atual:', status);
    
    try {
      // Usar o c√°lculo financeiro correto
      const calculatedResult = await calculation;
      
      console.log('üöÄ Navegando para resultado da simula√ß√£o com dados corretos:', calculatedResult);
      navigation.navigate('SimulationResult', {
        productId: route.params?.productId,
        amount: route.params?.amount,
        months: route.params?.months || 96,
        result: calculatedResult,
      });
    } catch (error) {
      console.error('‚ùå Erro no c√°lculo da simula√ß√£o:', error);
      
      // Fallback com valores b√°sicos em caso de erro
      const numericAmount = parseFloat(route.params?.amount?.replace(/\./g, '').replace(',', '.') || '0');
      const months = route.params?.months || 96;
      const rateMonthly = 0.0156; // Taxa de fallback
      
      const installment = numericAmount * rateMonthly / (1 - Math.pow(1 + rateMonthly, -months));
      const roundedInstallment = Math.round(installment * 100) / 100;
      const total = roundedInstallment * months;
      const totalInterest = total - numericAmount;
      
      const fallbackResult = {
        installment: roundedInstallment,
        total: total,
        totalInterest: totalInterest,
        rate: rateMonthly * 100,
        rateAnnual: (Math.pow(1 + rateMonthly, 12) - 1) * 100,
        amortizationType: 'PRICE' as const,
        schedule: []
      };
      
      navigation.navigate('SimulationResult', {
        productId: route.params?.productId,
        amount: route.params?.amount,
        months: months,
        result: fallbackResult,
      });
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.content}>
        {/* Apenas a anima√ß√£o centralizada */}
        <SimulationLottieAnimation
          isLoading={status === 'loading'}
          onFinish={handleAnimationFinish}
          size={400} // Aumentado para ocupar mais da largura da tela
          testID="simulation-loading-animation"
        />
      </View>
    </SafeAreaView>
  );
};

export default SimulationLoadingScreen;
