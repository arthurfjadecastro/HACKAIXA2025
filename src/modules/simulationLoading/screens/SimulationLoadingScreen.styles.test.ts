import { styles } from './SimulationLoadingScreen.styles';

describe('SimulationLoadingScreen Styles', () => {
  it('should export styles object', () => {
    expect(styles).toBeDefined();
    expect(typeof styles).toBe('object');
  });

  it('should have container style', () => {
    expect(styles.container).toBeDefined();
    expect(styles.container.flex).toBe(1);
    expect(styles.container.backgroundColor).toBe('#F4F8FB');
  });

  it('should have content style', () => {
    expect(styles.content).toBeDefined();
    expect(styles.content.flex).toBe(1);
    expect(styles.content.justifyContent).toBe('center');
    expect(styles.content.alignItems).toBe('center');
  });

  it('should match snapshot', () => {
    expect(styles).toMatchSnapshot();
  });
});
