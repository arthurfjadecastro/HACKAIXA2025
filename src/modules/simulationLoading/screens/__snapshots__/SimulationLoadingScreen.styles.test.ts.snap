// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimulationLoadingScreen Styles should match snapshot 1`] = `
{
  "container": {
    "backgroundColor": "#F4F8FB",
    "flex": 1,
  },
  "content": {
    "alignItems": "center",
    "flex": 1,
    "justifyContent": "center",
    "paddingHorizontal": 0,
  },
}
`;
